tensorflow/stream_executor/dnn.h

```diff
@@ -190,6 +190,15 @@ class RnnDescriptor {
   virtual ParamsRegions ParamsBiasRegions() const { return ParamsRegions(); }
 };
 
+// Specifies the CTC Loss computation.
+//
+// The user is responsible for releasing this descriptor when it is no longer
+// in use. The destructor releases the underlying descriptors.
+class CtcLossDescriptor {
```

Comment:
Instead of having a base class, can we simply have:
```
class CtcLossDescriptor {
 public:
  explicit CtcLossDescriptor(DataType);
  DataType GetComputeType() const;

 private:
  DataType comp_type_;
};
```

Reply:
I cannot follow the point here. I am following the style of the above RnnDescriptor example. My understanding is we have a device-agnostic base class XDescriptor in dnn.h, then we have specific CuDNNXDescriptor inheriting it in cuda_dnn.h/cc. I can see many other descriptors like RnnDescriptor or RnnSequenceTensorDescriptor did that.

Reply:
This file is not consistent in how to allow implementations customizing the data structure. For example, CovolutionDescriptor isn't virtual. I think we can start with a simple class, and turn it into virtual later.

Reply:
> This file is not consistent with itself in how to allow implementations customizing the data structure. For example, CovolutionDescriptor isn't virtual. I think we can start with a simple class, and turn it into virtual later.